core/src/main/java/com/scalar/db/transaction/consensuscommit/Snapshot.java

```diff
@@ -347,6 +348,18 @@ void toSerializableWithExtraRead(DistributedStorage storage)
     parallelExecutor.validate(tasks);
   }
 
+  private boolean isChanged(
+      Optional<TransactionResult> latestResult, Optional<TransactionResult> result) {
+    if (latestResult.isPresent() != result.isPresent()) {
+      return true;
+    }
+    if (!latestResult.isPresent()) {
+      return false;
+    }
+    return !latestResult.get().getId().equals(result.get().getId())
+        || latestResult.get().getVersion() != result.get().getVersion();
+  }
+
```

Comment:
This is the new comparison logic. It compares only `id` and `version` values of the read and re-read records.

Reply:
It's a super rare case, but a read record could be deleted, created, and updated several times to get to the read version by other transactions, so we need to compare the transaction ID as well?

Reply:
Thank you for taking a look at it!

Yes, I think the logic already compare the transaction ID (`!latestResult.get().getId().equals(result.get().getId())`). I think the logic compares both the transaction ID and version. That's not enough?
```java
    return !latestResult.get().getId().equals(result.get().getId())
        || latestResult.get().getVersion() != result.get().getVersion();
```

Reply:
Oh, sorry! I was a bit confused with Scalar DL. Thank you!